### PR TITLE
fix: make cornerRadius work for stacked bars with timeUnit

### LINKED ISSUE #5800: cornerRadius doesn't work for stacked bars with timeUnit
Failing spec:

```json
{
  "$schema": "https://vega.github.io/schema/vega-lite/v4.json",
  "description": "Google's stock price over time.",
  "data": {"url": "data/stocks.csv"},
  "mark": {"cornerRadiusTopLeft": 2, "cornerRadiusTopRight": 2, "type": "bar"},
  "encoding": {
    "x": {"field": "date", "type": "temporal", "timeUnit": "year"},
    "y": {"field": "price", "type": "quantitative", "aggregate": "average"},
    "color": {"field": "symbol", "type": "nominal"}
  }
}
```

Symptom: bars don't show up at all.
Note: If we remove the cornerRadius, the spec will work properly.

### PR DESCRIPTION
**Pull request by kanitw**
fix #5800

image

### CODE REVIEW

File: src/compile/mark/mark.ts [hunk 1]
```diff
@@ -174,12 +174,12 @@ function getStackGroups(model: UnitModel) {
       }
     }
 
+    const groupByField = model.fieldDef(model.stack.groupbyChannel);
     // For bin we have to add bin channels.
```
Comment (donghaoren) [inline src/compile/mark/mark.ts]: Consider moving this comment to before the if statement, and describe timeUnit?

File: examples/compiled/stacked_bar_count_corner_radius_config.vg.json [hunk 2]
```diff
@@ -48,7 +48,7 @@
         "facet": {
           "data": "source_0",
           "name": "stack_group_main",
-          "groupby": ["month_date"],
+          "groupby": ["month_date", "month_date_end"],
```
Comment (donghaoren) [inline examples/compiled/stacked_bar_count_corner_radius_config.vg.json]: For this particular spec, `month_date_end` is not being used, but I guess it doesn't hurt to add it there?

File: src/compile/mark/mark.ts (hunk 1 above)
Comment (kanitw) [reply src/compile/mark/mark.ts]: Why moving?   Unlike Vega codebase, we prefer to declare things where they are used.

File: examples/compiled/stacked_bar_count_corner_radius_config.vg.json (hunk 2 above)
Comment (kanitw) [reply examples/compiled/stacked_bar_count_corner_radius_config.vg.json]: Yeah, probably doesn't matter. We do generate this as an output from time unit earlier anyway.

File: src/compile/mark/mark.ts (hunk 1 above)
Comment (kanitw) [reply src/compile/mark/mark.ts]: (if you agree, please go ahead and merge. :D)

Comment (donghaoren) [reply src/compile/mark/mark.ts]: I thought the comment is describing the what the if statement is doing?

Comment (kanitw) [reply src/compile/mark/mark.ts]: oh you mean this if below.

**Comment by kanitw** [comment]
> I just found in the current version (before this PR), if I use "x": {"timeUnit": "month", "field": "date", "type": "ordinal"} it's going to work, but "x": {"timeUnit": "month", "field": "date", "type": "temporal"} is not going to work. It seems like bin is set differently based on whether the type is temporal or ordinal?

Yes, ordinal timeUnit would use ordinal scales rather than continuous scales. :)

**Pull request merged**